src/upload-tests.proj

```diff
@@ -77,6 +77,12 @@
     <HelixArchLabel>$(ArchGroup)</HelixArchLabel>
     <HelixConfigLabel>$(ConfigurationGroup)</HelixConfigLabel>
     <MaxRetryCount>1</MaxRetryCount>
+    <!-- Unique container name per build and configuration, e.g., corefx-beta-12345-01-windows-x64 -->
+    <!-- Azure Storage containers can only contain [a-z, 0-9, -] in their names and $(OSGroup) is "Windows_NT" on Windows... -->
+    <ProcessedOSGroup>$(OSGroup.ToLower())</ProcessedOSGroup>
+    <ProcessedOSGroup Condition="'$(TargetsWindows)' == 'true'">windows</ProcessedOSGroup>
+    <ProcessedArchGroup>$(ArchGroup.ToLower())</ProcessedArchGroup>
+    <ContainerName Condition="'$(ContainerName)' == '' And '$(IsOfficial)' != ''">corefx-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)-$(ProcessedOSGroup)-$(ProcessedArchGroup)-$(TargetGroup)</ContainerName>
```

Comment:
I think we should use the same naming convention for official builds so I'd suggest you just add a qualifier for official vs ci builds. There is also potentially a need to include the branch name because 2 different branches may end up building with the same build information.

Reply:
Actually why are you checking for `'$(IsOfficial)' != ''` what is that doing for us?

Reply:
`IsOfficial` is only set when `OfficialBuildId` is populated, which should only be during official builds from VSTS.  We only care about having a stable name when it is an official build, and not for every CI build or dev build.  I was under the impression that if stable names were turned on for unofficial builds, i.e., CI and dev builds, that there would be name conflicts for the container name.

Reply:
OK if it intended only for official builds then please update the condition to `'$(IsOfficial)' == 'true'` to make that clear.